dev/validate_bom__deploy.py

```diff
@@ -0,0 +1,392 @@
+    """Returns the sequence of Bash statements to fetch and run InstallHalyard.
+    """
+    options = self.options
+    script = []
+    script.append('curl -s -O {url}'.format(url=options.halyard_install_script))
+    install_params = ['-y']
+    if options.halyard_repository is not None:
+      install_params.extend(['--repository', options.halyard_repository])
+    if options.spinnaker_repository is not None:
+      install_params.extend(
+          ['--spinnaker-repository', options.spinnaker_repository])
+    script.append('sudo bash ./InstallHalyard.sh {install_params}'
+                  .format(install_params=' '.join(install_params)))
+
+    script.append('hal --color=false config deploy edit --type {type}'
+                  .format(type=options.deploy_spinnaker_type))
+    return script
+
+
+class GoogleValidateBomDeployer(BaseValidateBomDeployer):
+  """Concrete deployer used to deploy Hal onto Google Cloud Platform.
+
+  This class is not intended to be constructed directly. Instead see the
+  free function make_deployer() in this module.
+  """
+
+  EMPTY_SSH_KEY = os.path.join(os.environ['HOME'], '.ssh', 'google_empty')
+
+  def __init__(self, options):
+    super(GoogleValidateBomDeployer, self).__init__(options)
+
+  @classmethod
+  def init_platform_argument_parser(cls, parser):
+    """Adds custom configuration parameters to argument parser.
+
+    This is a helper function for the free function init_argument_parser().
+    """
+    parser.add_argument(
+        '--google_deploy_project',
+        default=None,
+        help='Google project to deploy to if --deploy_platform is "gce".')
+    parser.add_argument(
+        '--google_deploy_zone',
+        default='us-central1-f',
+        help='Google zone to deploy to if --deploy_platform is "gce".')
+    parser.add_argument(
+        '--google_deploy_instance',
+        default=None,
+        help='Google instance to deploy to if --deploy_platform is "gce".')
+
+  @classmethod
+  def validate_options_helper(cls, options):
+    """Adds custom configuration parameters to argument parser.
+
+    This is a helper function for make_deployer().
+    """
+    if not options.google_deploy_project:
+      raise ValueError('--google_deploy_project not specified.')
+    if not options.google_deploy_instance:
+      raise ValueError('--google_deploy_instance not specified.')
+
+  def make_port_forward_command(self, service, local_port, remote_port):
+    """Implements interface."""
+    options = self.options
+    return [
+        'gcloud', 'alpha', 'compute', 'ssh',
+        '--quiet', '--verbosity=none',
+        options.google_deploy_instance,
+        '--ssh-key-file={ssh_key}'.format(ssh_key=self.EMPTY_SSH_KEY),
+        '--project={project}'.format(project=options.google_deploy_project),
+        '--zone={zone}'.format(zone=options.google_deploy_zone),
+        '--', '-L {local_port}:localhost:{remote_port}'.format(
+            local_port=local_port, remote_port=remote_port)]
```

Comment:
Since you're using this just to open a connection, you can pass `-N` as well to avoid spawning a shell on the target machine.